### ISSUE TITLE
tpt pathway decomposition fails when requesting all pathways

### ISSUE THREAD

**Issue by franknoe** [markovmodel/PyEMMA#251]
In the BPTI notebook, when requesting all pathways by:

``` python
cgflux.pathways()
```

or 

``` python
cgflux.pathways(fraction=1.0)
```

this exception happens:

``` python
/Users/noe/anaconda/lib/python2.7/site-packages/pyEMMA-1.2-py2.7-macosx-10.5-x86_64.egg/pyemma/msm/flux/sparse/pathways.pyc in pathway(F, A, B)
    146                           "obtained from a reversible transition meatrix."))
    147     else:
--> 148         wL = pathway(F, A, [b1, ])
    149     if np.any(B == b2):
    150         wR = [b2, ]

/Users/noe/anaconda/lib/python2.7/site-packages/pyEMMA-1.2-py2.7-macosx-10.5-x86_64.egg/pyemma/msm/flux/sparse/pathways.pyc in pathway(F, A, B)
    138 
    139     """
--> 140     b1, b2, F = find_bottleneck(F, A, B)
    141     if np.any(A == b1):
    142         wL = [b1, ]

/Users/noe/anaconda/lib/python2.7/site-packages/pyEMMA-1.2-py2.7-macosx-10.5-x86_64.egg/pyemma/msm/flux/sparse/pathways.pyc in find_bottleneck(F, A, B)
     43 
     44     """Check if edge with largest conductivity connects A and B"""
---> 45     b = np.array(row[-1], col[-1])
     46     if has_path(b, A, B):
     47         return b

IndexError: index -1 is out of bounds for axis 0 with size 0
```

**Comment by marscher** [comment]
find_bottleneck does not handle the case of an empty flux matrix

**Comment by trendelkampschroer** [comment]
@franknoe: Could you point me to the location for the BPTI notebook and the resources required to run it. Starting from there I will try to find out what went wrong.

@marscher: Thanks for pointing out this problem in find_bottleneck. I will have a look at it too.

**Comment by franknoe** [comment]
I think you can just use the tpt ipython notebook because BPTI is not 
needed for that. If you ask the pathway decomposition to give the 
fraction 1.0 (all) of the pathways, then there is an exception.

Am 01/07/15 um 17:10 schrieb Benjamin Trendelkamp-Schroer:

> @franknoe https://github.com/franknoe: Could you point me to the 
> location for the BPTI notebook and the resources required to run it. 
> Starting from there I will try to find out what went wrong.
> 
> @marscher https://github.com/marscher: Thanks for pointing out this 
> problem in find_bottleneck. I will have a look at it too.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/markovmodel/PyEMMA/issues/251#issuecomment-117711158.

## 

---

Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de

## Mail:  Arnimallee 6, 14195 Berlin, Germany

**Comment by trendelkampschroer** [comment]
Okay will do that.

**Comment by trendelkampschroer** [comment]
This has been fixed by e37c082a2ea5251edd9c639889d9abffb7ef5fed

**Comment by marscher** [comment]
This issue was moved to markovmodel/msmtools#7
